pkg/networkmanager/ip-settings.jsx

```diff
@@ -132,21 +137,41 @@
             onClose: Dialogs.close,
         });
     };
-    const addressIpv4Helper = (address) => {
-        const config = { address, netmask: '', gateway: '' };
+
+    const addressIpv4Helper = (address, i) => {
+        const config = { address, prefix: '' };
         const split = address.split('.');
 
-        if (split.length !== 4)
+        if (split.length !== 4) {
+            if (i === 0 && !gatewaySetExplicitly)
+                setDefaultGateway("");
             return config;
+        }
 
-        config.gateway = `${split[0]}.${split[1]}.${split[2]}.${split[3] === "1" ? "254" : "1"}`;
         if (split[0] >= 0 && split[0] <= 127) {
-            return { ...config, netmask: "255.0.0.0" };
+            config.prefix = "255.0.0.0";
         } else if (split[0] >= 128 && split[0] <= 191) {
-            return { ...config, netmask: "255.255.0.0" };
+            config.prefix = "255.255.0.0";
         } else if (split[0] <= 192 && split[0] <= 223) {
-            return { ...config, netmask: "255.255.255.0" };
-        } else return { ...config, gateway: '' };
+            config.prefix = "255.255.255.0";
+        }
+
+        // pre-fill default gateway based on the first address for classfull prefixes
+        if (i === 0 && config.prefix !== "" && !gatewaySetExplicitly) {
+            setDefaultGateway(`${split[0]}.${split[1]}.${split[2]}.${split[3] === "1" ? "254" : "1"}`);
+        }
+
+        return config;
+    };
+
+    const removeAddress = (i) => {
+        // also reset gateway when removing the last address
+        if (addresses.length === 1) {
+            setDefaultGateway("");
+            setGatewaySetExplicitly(false);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22330-c3370d2a-20250917-072812-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)